tests/v1/sample/test_sampler.py

```diff
@@ -77,6 +77,46 @@ def _create_allowed_token_ids(
     return mask
 
 
+def _create_bad_words_token_ids(
+        batch_size: int, vocab_size: int,
+        bad_words_lengths: list[tuple[int]]) -> dict[int, list[list[int]]]:
+    bad_words_token_ids = {}
+    for batch_idx in range(batch_size):
```

Comment:
Maybe don't set this for every row in the batch here... it's hiding the bug in bad_words.py otherwise :)

Reply:
Added a random empty batch to this unit test